File: backend/app/schemas/restaurant.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RestaurantOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    cuisine: str
    rating: float
    image_url: str
    lat: float
    lng: float
    version: int
    updated_at: datetime
    deleted: bool = False


class MenuItemOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    restaurant_id: uuid.UUID
    name: str
    description: str
    price_minor: int
    currency: str
    image_url: str
    version: int
    updated_at: datetime
    deleted: bool = False


class RestaurantPage(BaseModel):
    items: list[RestaurantOut]
    next_cursor: str | None
    has_more: bool

File: android/core/model/src/main/kotlin/com/ordertracking/core/model/Restaurant.kt
package com.ordertracking.core.model

data class Restaurant(
    val id: String,
    val name: String,
    val cuisine: String,
    val rating: Double,
    val imageUrl: String,
    val lat: Double,
    val lng: Double,
)

data class MenuItem(
    val id: String,
    val restaurantId: String,
    val name: String,
    val description: String,
    val priceMinor: Long,
    val currency: String,
    val imageUrl: String,
)
